docs/technical-documentation/flysystem.md

```diff
@@ -0,0 +1,50 @@
+# Flysystem
+
+Islandora uses [Flysystem](https://github.com/thephpleague/flysystem) and the [associated Drupal module](https://www.drupal.org/project/flysystem) to persist binary files to Fedora instead of keeping a copy in both Drupal and Fedora.
+
+## Background
+
+"[Flysystem](https://flysystem.thephpleague.com/docs/) is a filesystem abstraction library for PHP" which allows applications to read from and write to a variety of data source beyond the local file system, such as an [SFTP server](https://flysystem.thephpleague.com/docs/adapter/sftp/), [Amazon S3](https://flysystem.thephpleague.com/docs/adapter/aws-s3/), and [Zip files](https://flysystem.thephpleague.com/docs/adapter/zip-archive/) provided an [Adapter](https://flysystem.thephpleague.com/docs/advanced/creating-an-adapter/) is available to support it. Flysystem Adapters extend a single class implementing `League\Flysystem\FilesystemInterface` although some separate adapter traits for common actions and properties, such as the StreamedCopyTrait, are available.
+
+The [Flysystem Drupal module](https://www.drupal.org/project/flysystem) extends Flysystem to work within the Drupal filesystem structure. Drupal flysystem plugins include a Flysystem adapter (if not provided by default or in another library) and a class implementing `Drupal\flysystem\Plugin\FlysystemPluginInterface` which instantiates the Flysystem adapter based on the Drupal site's configuration.
```

Comment:
You might also want to point out that Drupal's flysystem module uses `twistor/flysytem-stream-wrapper` to let you manipulate files using standard PHP stream functions no matter where they reside.